operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/DependentResource.java

```diff
@@ -5,8 +5,49 @@
 import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.javaoperatorsdk.operator.api.reconciler.Context;
 
+/**
+ * An interface to implement and provide dependent resource support.
+ *
+ * @param <R> the dependent resource type
+ * @param <P> the associated primary resource type
+ */
 public interface DependentResource<R, P extends HasMetadata> {
+
+  /**
+   * Reconciles the dependent resource given the desired primary state
+   *
+   * @param primary the primary resource for which we want to reconcile the dependent state
+   * @param context {@link Context} providing useful contextual information
+   * @return a {@link ReconcileResult} providing information about the reconciliation result
+   */
   ReconcileResult<R> reconcile(P primary, Context<P> context);
 
+  /**
+   * Retrieves the dependent resource associated with the specified primary one
+   *
+   * @param primaryResource the primary resource for which we want to retrieve the secondary
+   *        resource
+   * @return an {@link Optional} containing the secondary resource or {@link Optional#empty()} if it
+   *         doesn't exist
+   */
   Optional<R> getResource(P primaryResource);
+
+  /**
+   * Retrieves the resource type associated with this DependentResource
+   *
+   * @return the resource type associated with this DependentResource
+   */
+  Class<R> resourceType();
+
+  /**
+   * Computes a default name for the specified DependentResource class
+   *
+   * @param dependentResourceClass the DependentResource class for which we want to compute a
+   *        default name
+   * @return the default name for the specified DependentResource class
+   */
+  @SuppressWarnings("rawtypes")
+  static String defaultNameFor(Class<? extends DependentResource> dependentResourceClass) {
+    return dependentResourceClass.getCanonicalName();
```

Comment:
Could not we move this code to a new constructor to `DependetResourceSpec`. Isn't this related only to managed dependent resources?

Reply:
This isn't managed-specific: all dependent resources need to be registered with a name so if someone doesn't want to think of one, then they need to be able to compute the default one that will be used. So I think it makes sense to have that method here.

Reply:
> all dependent resources need to be registered with a name

Standalones should not. or?

Reply:
Standalone also need to be registered with a name. If you don't provide one, it will be generated for you, which is why you need to know what the default name is.

Reply:
where is the standaline registered for example in this example:

https://github.com/java-operator-sdk/java-operator-sdk/blob/7f82f54efe90caa6d3460039eb13eec00229fb1d/sample-operators/webpage/src/main/java/io/javaoperatorsdk/operator/sample/WebPageStandaloneDependentsReconciler.java#L33-L33

?

Reply:
Hmm, that's a good point… but I think that even standalone `DependentResource` implementations should be registered so that everything could be managed the same way. Probably outside of the scope for now, though. 

The thing that is annoying is that there is a more or less one-to-one relationship with the associated `EventSource` and in the non-managed way, this isn't well handled. Typically, the `EventSource` should have the same name than the associated `DependentResource` for less confusion but this isn't the case for standalone situations…

Reply:
The dependent resource - event source. Does not need to be one to one relationship. In standalone mode we can share event source now between more dependent resources. Like if someone wants to create more config maps, does not need more event source just one.
These optimizations could be done just in standalone mode. And that is right. Since all the minor optimizations could be done just there.

Reply:
But by default should have the same name I guess.

Reply:
That should be the case.

Reply:
And actually, if you share event sources between dependents then they cannot all have the same name, obviously.